### PR TITLE
feat(cache): support multiple versions of symfony/cache

### PR DESCRIPTION
**Pull request by Gounlaf**
Create a CacheProvider to create the appropriate instance of cache depending on symfony cache version.
This is to support versions ^3.3|^4.0|^5.0 of symfony/cache (since 4.3, classes in Simple namespace are deprecated, and removed in 5.x)

### CODE REVIEW

**Comment by natebrunette** [comment]
Part of this already got merged in, you might want to rebase.

**Comment by Gounlaf** [comment]
> Part of this already got merged in, you might want to rebase.

Hi @natebrunette,

It's done on tebru/gson-php, not here.
I've implemented it here

**Comment by Gounlaf** [comment]
Hi @natebrunette,

Can you take a look?
This update is necessary to use `tebru/retrofit-php` with `tebru/retrofit-php-converter-gson` / `tebru/gson-php` with symfony components 5.*

Thanks

**Comment by natebrunette** [comment]
@Gounlaf it looks like tests failed

File: src/Internal/CacheProvider.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use Symfony\Component\Cache\Simple\ChainCache;
+use Symfony\Component\Cache\Simple\NullCache;
+use Symfony\Component\Cache\Simple\PhpFilesCache;
+
+/**
+ * @codeCoverageIgnore
+ */
+final class CacheProvider
+{
+    public static function createFileCache(string $cacheDir): CacheInterface
```
Comment (natebrunette) [inline src/Internal/CacheProvider.php:32]: docblock

**Review by natebrunette — CHANGES_REQUESTED** [review]
Could you just remove the lock file as well?

**Comment by natebrunette** [comment]
Once the issues are address and it's squashed I'll merge

File: src/Internal/CacheProvider.php (hunk 1 above)
Comment (Gounlaf) [reply src/Internal/CacheProvider.php:32]: fixed

**Comment by natebrunette** [comment]
@Gounlaf shouldn't there be an update to gitignore too?

**Comment by Gounlaf** [comment]
> @Gounlaf shouldn't there be an update to gitignore too?

Done; if it's okay I will squash again

**Comment by natebrunette** [comment]
probably add a `/` to make it `/composer.lock` to be safe, but yeah, that's fine.

**Comment by Gounlaf** [comment]
> probably add a `/` to make it `/composer.lock` to be safe, but yeah, that's fine.

fixed & squashed

**Comment by Gounlaf** [comment]
Hi @natebrunette,

Since it's merged, can you please make a release?

Thanks,
Regards.

**Comment by natebrunette** [comment]
I tagged new releases of gson and retrofit

**Pull request merged**